examples/specs/normalized/layer_line_errorband_pre_aggregated_normalized.vl.json

```diff
@@ -29,7 +29,8 @@
             "x": {
               "field": "year_Year",
               "type": "temporal",
-              "title": "Year (year)"
+              "title": "Year (year)",
+              "timeUnit": "year"
```

Comment:
We cannot only have `timeUnit` in the encoding because `timeUnit` is required for `groupby` in aggregate for parts of composite mark like this:
```json
"transform": [
  {
    "timeUnit": "month",
    "field": "date",
    "as": "month_date"
  },
  {
    "aggregate": [
      {
        "op": "max",
        "field": "temp_max",
        "as": "max_temp_max"
      }
    ],
    "groupby": [
      "month_date"
    ]
  }
]
```

So, I think our only choice is to format the field for different type of timeUnit?

Reply:
Right. But you can still specify a format. Is the problem that for ordinal we don't have a way to specity time formats?

Reply:
Yes, I think we can have another calculate transform to format the field, then use the new field instead?

Reply:
I don't understand the suggestion. Can you explain it?

Reply:
In the normalized spec, instead of only transform `timeUnit` like this:
```json
"transform": [
  {
    "timeUnit": "month",
    "field": "date",
    "as": "month_date"
  },
  {
    "aggregate": [
      {
        "op": "max",
        "field": "temp_max",
        "as": "max_temp_max"
      }
    ],
    "groupby": [
      "month_date"
    ]
  }
]
```
should we format them first like this?
```json
"transform": [
  {
    "timeUnit": "month",
    "field": "date",
    "as": "tmp_month_date"
  },
  {
    "calculate": "['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'][month(datum.tmp_month_date)]",
    "as": "month_date"
  },
  {
    "aggregate": [
      {
        "op": "max",
        "field": "temp_max",
        "as": "max_temp_max"
      }
    ],
    "groupby": [
      "month_date"
    ]
  }
]
```

Reply:
No, we should set the format on the endcoding. I think at least that is the easiest way forward. Cc @kanitw what do you think?